### PR TITLE
ZNTA-1081 Use @Synchronized for @SessionScoped beans

### PR DESCRIPTION
**Pull request by seanf**
https://zanata.atlassian.net/browse/ZNTA-1081

### CODE REVIEW

File: zanata-war/src/main/java/org/zanata/action/RoleSearch.java [hunk 1]
```diff
@@ -4,20 +4,23 @@
...
 import org.zanata.security.ZanataIdentity;
+import org.zanata.util.Synchronized;
 
 /**
  * @author Patrick Huang
  *         <a href="mailto:pahuang@redhat.com">pahuang@redhat.com</a>
  */
 @Named("zanataRoleSearch")
 // TODO this should probably be ViewScoped or even RequestScoped (plus xhtml changes)
-@javax.enterprise.context.SessionScoped
```
Comment (djansen-redhat) [inline zanata-war/src/main/java/org/zanata/action/RoleSearch.java]: Is this comment related to the altered code?

File: zanata-war/src/main/java/org/zanata/action/UserSearch.java [hunk 2]
```diff
@@ -3,21 +3,23 @@
...
 import org.zanata.security.annotations.CheckRole;
+import org.zanata.util.Synchronized;
 
 /**
  * @author Patrick Huang <a
  *         href="mailto:pahuang@redhat.com">pahuang@redhat.com</a>
  */
 @Named("userSearch")
 // TODO this should probably be ViewScoped or even RequestScoped (with changes in xhtml)
-@javax.enterprise.context.SessionScoped
```
Comment (djansen-redhat) [inline zanata-war/src/main/java/org/zanata/action/UserSearch.java]: Is this comment related to the altered code?

File: zanata-war/src/main/java/org/zanata/seam/security/ZanataJpaIdentityStore.java [hunk 3]
```diff
@@ -106,13 +106,12 @@ public boolean apiKeyAuthenticate(String username, String apiKey) {
...
     /**
      * Custom authentication that ignores the account's enabled state.
-     *
      * @param username
      * @param password
      * @return
      * @see {@link ZanataJpaIdentityStore#authenticate(String, String)}
      */
-    public boolean authenticateEvenIfDisabled(String username, String password) {
+    public boolean checkPasswordIgnoringActivation(String username, String password) {
```
Comment (djansen-redhat) [inline zanata-war/src/main/java/org/zanata/seam/security/ZanataJpaIdentityStore.java]: Didn't like the name?

**Comment by djansen-redhat** [comment]
@seanf This seems to break Yahoo login.

File: zanata-war/src/main/java/org/zanata/action/RoleSearch.java (hunk 1 above)
Comment (seanf) [inline zanata-war/src/main/java/org/zanata/action/RoleSearch.java]: Yes, someone still need to look at this and decide whether SessionScoped is the best option.

File: zanata-war/src/main/java/org/zanata/action/UserSearch.java (hunk 2 above)
Comment (seanf) [inline zanata-war/src/main/java/org/zanata/action/UserSearch.java]: Ditto.

File: zanata-war/src/main/java/org/zanata/seam/security/ZanataJpaIdentityStore.java (hunk 3 above)
Comment (seanf) [inline zanata-war/src/main/java/org/zanata/seam/security/ZanataJpaIdentityStore.java]: It wasn't really authenticating the user (in terms of changing the state of the user session), so the name was misleading.

**Comment by seanf** [comment]
@djansen-redhat How does it break? Does it cause a 30 second deadlock/timeout during authentication? Any error messages?

**Comment by seanf** [comment]
@djansen-redhat Never mind; I just saw your email.

**Comment by carlosmunoz** [comment]
@huangp could I please get you to review this one?

**Comment by huangp** [comment]
@carlosmunoz I made the last commit so don't think I should review it ;) I've asked @seanf to review the last commit

**Comment by seanf** [comment]
You have? When was that?

I'll have a look. Have you reviewed my earlier commits?

On 17 May 2016 at 13:01, Patrick Huang <notifications@github.com> wrote:

> @carlosmunoz <https://github.com/carlosmunoz> I made the last commit so
> don't think I should review it ;) I've asked @seanf
> <https://github.com/seanf> to review the last commit
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/1158#issuecomment-219607606>
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat

File: zanata-model/src/main/java/org/zanata/model/validator/UniqueValidator.java [hunk 4]
```diff
@@ -55,6 +57,10 @@ public boolean isValid(Object value, ConstraintValidatorContext context) {
     }
 
     private int countRows(Object value) {
+        // we need to use entityManager.unwrap because  injected session will
+        // be a weld proxy and criteria.getExecutableCriteria method will try
+        // to cast it to SessionImplementor (ClassCastException)
+        Session session = entityManager.unwrap(Session.class);
```
Comment (seanf) [inline zanata-model/src/main/java/org/zanata/model/validator/UniqueValidator.java]: @huangp Glad you got to the bottom of it. I think we should deprecate the session producer in EntityManagerProducer with a TODO comment warning about this problem, and maybe get rid of it in `master` in a separate PR.

**Comment by huangp** [comment]
@seanf your changes reviewed and added the comment

**Comment by seanf** [comment]
:+1: on changes by @huangp

**Comment by djansen-redhat** [comment]
:white_check_mark: Verified that @huangp's change  fixes the Yahoo issue.
